pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -0,0 +1,85 @@
+package fleetresourcehandler
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"go.goms.io/fleet/pkg/webhook/validation"
+)
+
+const (
+	// ValidationPath is the webhook service path which admission requests are routed to for validating custom resource definition resources.
+	ValidationPath = "/validate-v1-fleetresourcehandler"
+	groupMatch     = `^[^.]*\.(.*)`
+)
+
+// Add registers the webhook for K8s bulit-in object types.
+func Add(mgr manager.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &fleetResourceValidator{Client: mgr.GetClient()}})
+	return nil
+}
+
+type fleetResourceValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func (v *fleetResourceValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+	var response admission.Response
+	if req.Operation == admissionv1.Create || req.Operation == admissionv1.Update || req.Operation == admissionv1.Delete {
+		switch req.Kind.String() {
+		case retrieveCRDGVK():
+			klog.V(2).InfoS("handling CRD resource", "GVK", retrieveCRDGVK())
+			response = v.handleCRD(req)
+		default:
+			klog.V(2).InfoS("resource is not monitored by fleet resource validator webhook", "GVK", req.Kind.String())
+			response = admission.Allowed("")
```

Comment:
Though this message is probably not read by anyone :D, it might still be better to have a short description here